test/test_intensify/test_hyperband.py

```diff
@@ -50,87 +51,89 @@ def setUp(self):
 
-        self.rh.add(config=self.config1, cost=1, time=1,
-                    status=StatusType.SUCCESS, instance_id=1,
-                    seed=0,
-                    additional_info=None)
+        # next HB stage
+        intensifier._update_stage()
 
-        inc, _ = intensifier.intensify(challengers=[self.config2],
-                                       incumbent=self.config1,
-                                       run_history=self.rh,
-                                       aggregate_func=average_cost)
+        self.assertEqual(intensifier.s, 2.0)
+        self.assertEqual(intensifier.hb_iters, 0)
+        self.assertEqual(intensifier.sh_intensifier.initial_budget, 0.25)
+        self.assertEqual(intensifier.sh_intensifier.n_configs_in_stage.tolist(), [4.0, 2.0, 1.0])
 
-        self.assertEqual(intensifier.s_max, 3)
-        self.assertEqual(inc, self.config1)
+        # HB iteration completed
+        intensifier.s = 0.0
+        intensifier._update_stage()
+
+        self.assertEqual(intensifier.s, intensifier.s_max)
+        self.assertEqual(intensifier.hb_iters, 1)
+        self.assertEqual(intensifier.sh_intensifier.initial_budget, 0.125)
+        self.assertEqual(intensifier.sh_intensifier.n_configs_in_stage.tolist(), [8.0, 4.0, 2.0, 1.0])
 
     @attr('slow')
-    def test_intensify_2(self):
+    def test_eval_challenger(self):
         """
-           test intensify for challengers used in iterations
+            since hyperband uses eval_challenger and get_next_challenger of the internal successive halving,
+            we don't test these method extensively
         """
 
         def target(x):
-            return (2*x['a'] + x['b'] + 1) / 1000.
+            return 0.1
 
-        stats = Stats(scenario=self.scen)
-        stats.start_timing()
-        taf = ExecuteTAFuncDict(ta=target, stats=stats, run_obj="quality")
-        taf.runhistory = RunHistory(aggregate_func=average_cost)
-        taf.runhistory.overwrite_existing_runs = True
+        taf = ExecuteTAFuncDict(ta=target, stats=self.stats)
+        taf.runhistory = self.rh
 
         intensifier = Hyperband(
-            tae_runner=taf, stats=taf.stats,
-            traj_logger=TrajLogger(output_dir=None, stats=taf.stats),
+            tae_runner=taf, stats=self.stats,
+            traj_logger=TrajLogger(output_dir=None, stats=self.stats),
             rng=np.random.RandomState(12345), deterministic=True, run_obj_time=False,
-            instances=[1], initial_budget=0.5, max_budget=1, eta=2)
+            instances=[None], initial_budget=0.5, max_budget=1, eta=2)
 
-        # ensuring correct parameter initialization
+        self.assertFalse(hasattr(intensifier, 's'))
+
+        # Testing get_next_challenger - get next configuration
+        config = intensifier.get_next_challenger(challengers=[self.config2], chooser=None, run_history=self.rh)
         self.assertEqual(intensifier.s, intensifier.s_max)
+        self.assertIsInstance(config, Configuration)
```

Comment:
You could check that the exact same configuration is returned. Alternatively, you could mock out successive halving and check that it is called correctly.